backend/src/nodes/nodes/pytorch/load_model.py

```diff
@@ -51,11 +52,18 @@ def run(self, path: str) -> Tuple[PyTorchModel, str, str]:
 
         try:
             logger.debug(f"Reading state dict from path: {path}")
-            state_dict = torch.load(
-                path,
-                map_location=torch.device(exec_options.full_device),
-                pickle_module=RestrictedUnpickle,  # type: ignore
-            )
+
+            if os.path.splitext(path)[1] == ".pt":
```

Comment:
is there a way we can check that it's a torchscript file without the extension? Technically there's nothing preventing anyone from saving a normal model as .pt or a JIT one as .pth, so a more robust check based on the file content would be much better imo

Reply:
I do not know. Seems kind of hard to do that without opening it first.

Reply:
I can think of a couple things. One is to attempt torch load, and then try jit load if it errors. The other is regex. TorchScript models have some extra files and folders that pth files don't have, including an extra pkl file (constants.pkl). Searching that string might be enough.

Reply:
Nevermind, loading a torchscript with torch load only fails if an operation isn't implemented, otherwise it's just a warning.

Reply:
This is fine then

Reply:
It will *probably* be fine. We can always revisit it if we have to, but I really don't know what to do other than read the file and regex it.